### PR TITLE
Revenue Additions

### PR DESCRIPTION
**Pull request by abreaux26**
**What did you change?**
Created total_revenue_by_date and top_revenue_earners

**Next Steps**
~~Merge Alex's PR first (literally went in so many circles, so built off of Alex's phenomenal work)~~
~~Rename variable names~~
~~Refactor and create helper methods~~

**Known Issues**
~~Won't work unless Alex's PR is merged first.~~

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -132,6 +132,25 @@ def invoice_total(invoice_id)
...
+    end
+  end
+
+  def top_revenue_earners(x=20)
+    hash = {}
+    invoices_by_merchant.each do |merchant, invoices|
+      hash[merchant] = revenue_by_merchant(merchant.id)
+    end
+
+    sorted = hash.sort_by do |merchant, revenue|
```
[the remark this answers is not in the document]
Comment (abreaux26) [reply lib/sales_analyst.rb:147]: going to fix this after all other PRs are merged

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -132,6 +132,25 @@ def invoice_total(invoice_id)
...
 
+  def total_revenue_by_date(date)
+    @engine.invoices_on(date).sum do |invoice|
+      invoice_total(invoice.id)
+    end
+  end
+
+  def top_revenue_earners(x=20)
+    hash = {}
+    invoices_by_merchant.each do |merchant, invoices|
```
[the remark this answers is not in the document]
Comment (abreaux26) [reply lib/sales_analyst.rb:143]: going to fix this after all other PRs are merged

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -132,6 +132,29 @@ def invoice_total(invoice_id)
...
+
+  def merchants_with_revenue
+    invoices_by_merchant.each_with_object({}) do |merchant_invoices, hash|
+      merchant = merchant_invoices[0]
+      hash[merchant] = BigDecimal(revenue_by_merchant(merchant.id), 6)
+    end
+  end
+
+  def merchants_sorted_by_revenue
+    merchants_with_revenue.sort_by do |_, revenue|
```
Comment (jbranham1) [inline lib/sales_analyst.rb:153]: What is the underscore doing?

Comment (abreaux26) [reply lib/sales_analyst.rb:153]: @jbranham1 that was a hound fix suggestion! It basically means when you declare two block variables and are not using one of them in the block of code, you can just put an underscore as a placeholder.

**Review by jbranham1 — APPROVED** [review]
Excited to push this out!

**Pull request merged**